src/AppBundle/Repository/CategoryRepository.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace AppBundle\Repository;
+
+use AppBundle\Entity\Category;
+use Faker;
+
+/**
+ * CategoryRepository.
+ *
+ * This class was generated by the Doctrine ORM. Add your own custom
+ * repository methods below.
+ */
+class CategoryRepository extends \Doctrine\ORM\EntityRepository
+{
+    /**
+     * Get random category.
+     *
+     * @return Category
+     */
+    public function getRandomCategory()
+    {
+        $count = $this->createQueryBuilder('c')
+            ->select('count(c.id)')
+            ->getQuery()
+            ->getSingleScalarResult();
+        $faker = Faker\Factory::create();
+        $random = $faker->numberBetween($min = 1, $max = $count);
+
+        $query = $this->createQueryBuilder('c')
+            ->where('c.id = :id')
+            ->setParameter('id', $random)
+            ->getQuery();
+
+        $category = $query->setMaxResults(1)->getOneOrNullResult();
```

Comment:
if your storage has a one or more entities, you code should return the entity, but this method can return null, if storage is not empty